subprojects/shipkit/src/main/groovy/org/shipkit/internal/comparison/ZipComparator.java

```diff
@@ -58,15 +68,15 @@ Diff compareZips(File previousFile, File currentFile) {
             }
             try {
                 current.remove(name);
-                if (!streamsEqual(file1.getInputStream(file1.getEntry(name)), file2.getInputStream(file2
-                        .getEntry(name)))) {
+                if (!streamsEqual(previousFile.getInputStream(previousFile.getEntry(name)),
+                                  currentFile.getInputStream(currentFile.getEntry(name)))) {
                     changedFiles.add(name);
```

Comment:
This is a good place to notify some listener that a file has changed between source jars. The listener receives info: file name and 2 contents. The listener can check if the file is interesting (e.g. 'dependency-info.json') and start building a nice diff.

Next time we can do it this way :) I think it would simplify the algorithm and avoid redundant work. This way, we could avoid the "previous.remove(ComparePublications.DEPENDENCY_INFO_FILEPATH);" code.

Reply:
I added TODO in code addressed at me. Also for unit tests for ComparePublications class.